pkg/store/storepb/rpc.proto

```diff
@@ -122,6 +122,9 @@ message SeriesRequest {
   // shard_info is used by the querier to request a specific
   // shard of blocks instead of entire blocks.
   ShardInfo shard_info = 13;
+
+  // Control whether the store should calculate the checksum/hash of chunks
+  bool calculate_chunk_checksums = 14;
```

Comment:
Should we remove this and control that in the implementations (each server option for all requests). I don't see a need for request to ask for it explicitly 🤔

Reply:
I think it's because of https://github.com/thanos-io/thanos/pull/5703#discussion_r974439342

Reply:
On the Thanos side I think we can always enable it. But Cortex also relies on the store gateway and the store API and calculating hash on the cortex side is unnecessary for now

Reply:
Please let me know what is best here. :)

Reply:
I think it's fine to have it as a command-line argument instead of adding it to the request itself. That's what @bwplotka had in mind I believe.

Reply:
Umm yeah, sounds like a flag on store gateway is better

Reply:
Okay, I will add flag. But then we need flag for:

* Receiver (MultiTSDB -> TSDB)
* Ruler (TSDB)
* Store Gateway (Bucket Store)
* Sidecar (Prometheus Store)

Is that right?

Reply:
If we don't need to change the protobuf file, then seems it is also okay to not add any flags.
Since Cortex only shares the store gateway with Thanos, so as long as the `blockSeries` method exposes a boolean param to control not calculate hash then it is fine. On the thanos side, we can make it always `true`.

Reply:
If I got this right, we would need still to let them pass an option here: https://github.com/cortexproject/cortex/blob/d4fceb1172fff9ac332e525086d07f2ef182aefc/pkg/storegateway/bucket_stores.go#L484, right? Should I create a field on the bucket store to set wether or not to calculate hashes? Then, they could write a `BucketStoreOption` func that allows them to set this field. Let me know if this is okay.

Reply:
Sorry for the late reply. For me I think it is fine to add it at `blockSeries` to control that option. Do you think it works?

Reply:
I did that, I just thought that since that is a private function of the package they were not able to call it themselves.

Reply:
Ah you are right sorry. So we need either:
1. an option at BucketStore to turn on/off chunk hashing
2. modify series request

Reply:
Can you please recheck @yeya24 , I went with my idea to create a `BucketStoreOption` and enabled it by default on thanos side.